src/test/java/net/snowflake/client/core/arrow/DateConverterTest.java

```diff
@@ -162,4 +178,85 @@ public void testRandomDates() throws SFException {
       }
     }
   }
+
+  @Test
+  public void testTimezoneDates() throws SFException {
+    int testDay = 16911;
+    Map<String, String> customFieldMeta = new HashMap<>();
+    customFieldMeta.put("logicalType", "DATE");
+    // test normal date
+    FieldType fieldType =
+        new FieldType(true, Types.MinorType.DATEDAY.getType(), null, customFieldMeta);
+
+    DateDayVector vector = new DateDayVector("date", fieldType, allocator);
+
+    vector.setSafe(0, testDay);
+
+    // Test JDBC_FORMAT_DATE_WITH_TIMEZONE=TRUE with different session timezones
+    TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
+    String tz = System.getProperty("user.timezone");
+    ArrowVectorConverter converter = new DateConverter(vector, 0, this, true);
+    converter.setUseSessionTimezone(true);
+    converter.setSessionTimeZone(TimeZone.getTimeZone(tz));
+    Object obj = converter.toObject(0);
+    DateTimeConverter jsonConverter =
+        new DateTimeConverter(
+            TimeZone.getTimeZone(tz), this.getSession(), 0, true, false, true, true);
+    Date jsonDate =
+        jsonConverter.getDate(Integer.toString(testDay), 91, 91, TimeZone.getTimeZone(tz), 0);
+    Object utcObj =
+        ArrowResultUtil.getDate(testDay, TimeZone.getTimeZone("UTC"), TimeZone.getTimeZone(tz));
+
+    switch (tz) {
```

Comment:
in unit tests can we remove this condition logic or/and write also unit test for some timezones explicitly?